rosette-scala/src/main/scala/coop/rchain/rosette/Location.scala

```diff
@@ -149,90 +169,231 @@ object Location {
+            LTAddrVariable(indirect, level + adjustment, offset)
+
+          case LTBitField(indirect, level, offset, spanSize, sign) =>
+            LTBitField(indirect, level + adjustment, offset, spanSize, sign)
+
+          case _ => {
+            suicide(s"Location.adjustLevel: $loc")
+            null
+          }
+        })
+    }
+
+  def CtxtReg(n: Int): Location = {
+    if (n >= NumberOfCtxtRegs) {
+      suicide(s"Location.CtxtReg: invalid ctxt register ($n)");
+      null
+    }
+
+    LocationGT(LTCtxtRegister(n))
+  }
+
+  def ArgReg(n: Int): Location = {
+    if (n > MaxArgs) {
+      suicide(s"Location.ArgReg: invalid arg register index ($n)")
+      null
     }
+
+    LocationGT(LTArgRegister(n))
+  }
+
+  def LexVar(level: Int, offset: Int, indirect: Int): Location = {
+    if (level >= (1 << LexLevelSize) || offset >= (1 << LexOffsetSize)) {
+      val offsetStr = if (indirect != 0) s"($offset)" else s"$offset"
+      suicide(
+        s"Location.LexVar: unrepresentable location (lex[$level,$offsetStr])")
+      null
+    }
+    LocationGT(LTLexVariable(if (indirect == 0) 0 else 1, level, offset))
+  }
+
+  def AddrVar(level: Int, offset: Int, indirect: Int): Location = {
+    if (level >= (1 << AddrLevelSize) || offset >= (1 << AddrOffsetSize)) {
+      val offsetStr = if (indirect != 0) s"($offset)" else s"$offset"
+      suicide(
+        s"Location.AddrVar: unrepresentable location (addr[$level,$offsetStr])")
+      null
+    }
+    LocationGT(LTAddrVariable(if (indirect == 0) 0 else 1, level, offset))
+  }
+
+  def GlobalVar(n: Int): Location = {
+    if (n >= (1 << GlobalOffsetSize)) {
+      suicide(s"Location.GlobalVar: unrepresentable location (global[$n])")
+      null
+    }
+
+    LocationGT(LTGlobalVariable(n))
+  }
+
+  def BitField(level: Int,
+               offset: Int,
+               span: Int,
+               indirect: Int,
+               sign: Int): Location = {
+    if (level >= (1 << BitFieldLevelSize)
+        || offset >= (1 << BitFieldOffsetSize)
+        || span > (1 << BitFieldSpanSize)) {
+      val offsetStr = if (indirect != 0) s"($offset)" else s"$offset"
+      val signStr = if (sign == 0) "u" else "s"
+      suicide(
+        s"Location.BitField: unrepresentable location (${signStr}fld[$level,$offsetStr,$span])")
+      null
+    }
+
+    LocationGT(
+      LTBitField(indirect,
+                 level,
+                 offset,
+                 span % (1 << BitFieldSpanSize),
+                 if (sign == 0) 0 else 1))
+  }
+
+  def BitField00(offset: Int, span: Int, sign: Int): Location = {
+    if (offset >= (1 << BitField00OffsetSize)
+        || span > (1 << BitField00SpanSize)) {
+      val signStr = if (sign == 0) "u" else "s"
+      suicide(
+        s"Location.BitField: unrepresentable location (${signStr}fld[$offset,$span])")
+      null
+    }
+
+    LocationGT(
+      LTBitField00(offset,
+                   span % (1 << BitFieldSpanSize),
```

Comment:
Should be `BitField00SpanSize`.

Reply:
Without really thinking it through: Seems wrong since in the C++ implementation it is either `0` or `1`.

Reply:
Sorry about the merge; since I branched this one from dev-stay-location2, which was merged into dev, but wanted it to eventually get into dev itself, I wasn't sure how to go about that.  I switched the source of the comparison to dev, but also made a mistake with git remote that I had to fix.

Regarding line 389, the C++ says
```SET_BITFIELD00_SPAN(loc, ( span == (1 << BitField00SpanSize) ? 0 : span));```
which is the same as `span % (1 << BitField00SpanSize)` for the range 1 to (1 << BitField00SpanSize).

Reply:
Ah sorry, I see: we are checking for that range above.
I'll approve once you change line `389` to `BitField00SpanSize`.